parser/src/restparser.ts

```diff
@@ -0,0 +1,74 @@
+import { parse as pathParse } from "path";
+import { RestAnnotation } from "./ast";
+
+export function parseRestAnnotation(text: string) {
+    const fragments = text.split(" ");
+    const method = fragments[0].toUpperCase();
+
+    if (!["GET", "POST", "PUT", "DELETE", "PATCH"].includes(method)) {
+        throw new Error(`Unsupported method '${method}'`);
+    }
```

Comment:
Now I think we shouldn't throw an error, just pring a warning, [since the HTTP protocol specification doesn't restrict the possible methods, just declares some standards](https://tools.ietf.org/html/rfc7231#section-4.1).

Reply:
But we are supporting swagger and ["OpenAPI 3.0 supports get, post, put, patch, delete, head, options, and trace."](https://swagger.io/docs/specification/paths-and-operations/). Also, it is quite unlikely that someone decides to use a different HTTP method here. Most likely the person mistyped something. For now I believe it is better to wait for some real world use case (someone opening an issue complaining about this) before relaxing. What do you think?

Reply:
I'm fine with that then.